src/pages/index.js

```diff
@@ -71,24 +119,88 @@ addCardAddButton.addEventListener("click", () => {
 
 updateProfileButton.addEventListener("click", () => {
   const userData = userInfo.getUserInfo();
-  profileHeadingInput.value = userData.userName;
-  profileDescriptionInput.value = userData.userDescription;
+  profileHeadingInput.value = userData.name;
+  profileDescriptionInput.value = userData.about;
   editProfileFormValidator.resetValidation();
   updateProfilePopup.open();
 });
 
+updateAvatarButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
 // ! ||--------------------------------------------------------------------------------||
-// ! ||                                 Event Handlers                                 ||
+// ! ||                                 Functions                                 ||
 // ! ||--------------------------------------------------------------------------------||
+function handleValidation(form) {
+  form.enableValidation();
+}
+
 function handleAddCardFormSubmit(data) {
-  const card = renderCard(data);
-  cardSection.addItem(card);
+  this.showButtonProgress(true);
+  api
+    .addNewCard(data.name, data.link)
+    .then((res) => {
+      const card = renderCard(res);
+      cardSection.addItem(card);
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    });
+    this.showButtonProgress(false);
 }
 
 function handleImageClick(name, link) {
   newImagePopup.open(name, link);
 }
 
 function handleProfileSubmit(userData) {
-  userInfo.setUserInfo(userData);
-}
\ No newline at end of file
+  this.showButtonProgress(true);
+  api
+    .updateUserInfo(userData.name, userData.about)
+    .then((user) => {
+      userInfo.setUserInfo(user);
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    });
+    this.showButtonProgress(false);
+}
+
+function handleDeleteCard(_id) {
+  deletePopup.open();
+  deletePopup.setSubmitAction(() => {
+    api
```

Comment:
Needs correction: it's also needed to notify the user about loading process by changing the submit button text

Reply:
This has been corrected.